providers/downloadPluginBinary.go

```diff
@@ -0,0 +1,38 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func DownloadPluginBinaryFactory(name, version string) ProviderFactory {
+	factory := func(ctx context.Context) (int, error) {
+		binaryPath, err := DownloadPluginBinary(name, version)
+		if err != nil {
+			return 0, err
+		}
+		return startLocalBinary(ctx, binaryPath, name)
+	}
+	return factory
+}
+
+func DownloadPluginBinary(name, version string) (string, error) {
+	cmd := exec.Command("pulumi", "plugin", "install", "resource", name, version)
```

Comment:
Eh this one is a bit interesting. It uses global plugin cache which is reasonable. One day perhaps allow user to seal off the local plugin cache for the tests. Can do later.